### PR TITLE
[Task]: Deprecate HeadlessChrome processor

### LINKED ISSUE #14511: [Task]: Deprecate HeadlessChrome
### Task description

HeadlessChrome will be removed in Pimcore 11.x. Check https://github.com/pimcore/pimcore/pull/14457
We need to deprecate the whole class in 10.6 and add it to the upgrade notes

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/14511

### CODE REVIEW

**Comment by kingjia90** [comment]
Mhh, feels weird, theoretically the Chromium processor in https://github.com/pimcore/pimcore/pull/14457 should be introduced in 10.6 but that conflicts with the rule that 10.6 won't introduce new improvements 🙈

**Comment by brusch** [comment]
Since this processor is less a coding thing but a config, I think it would be ok, to do this during the upgrade process. 
WDYT?

**Comment by kingjia90** [comment]
> Since this processor is less a coding thing but a config, I think it would be ok, to do this during the upgrade process. WDYT?

mhh, that's true, just felt weird that there are no actual preparatory action that can be made in advance and keeps one in "pain" with relatively frequent deprecation messages that cannot be silenced.

One idea that came up is like a dummy config settings called `Chromium`  which under the hood runs the same as HeadlessChrome code but without all deprecation triggers. This could work as long as the configs are the same (which at some extent they are) or have a sort of FC layer for it, but might be an overkill, would rather directly add the actual new processor

File: lib/Web2Print/Processor/HeadlessChrome.php [hunk 1]
```diff
@@ -27,10 +27,22 @@
...
 
+/**
+ * @deprecated will be removed in Pimcore 11 and replaced by Processor\Chromium
+ */
 class HeadlessChrome extends Processor
 {
     private $nodePath = '';
 
+    public function __construct(){
+        trigger_deprecation(
```
Comment (brusch) [inline lib/Web2Print/Processor/HeadlessChrome.php]: @kingjia90 I wouldn't issue the deprecation message here, for the reasons you mentioned. I thin kit's enough if we document the fact that's going away in v11 (which is already done).

**Review by brusch — DISMISSED** [review]

File: lib/Web2Print/Processor/HeadlessChrome.php (hunk 1 above)
Comment (kingjia90) [reply lib/Web2Print/Processor/HeadlessChrome.php]: Thank you for your suggestion, looks much better now

**Review by mattamon — CHANGES_REQUESTED** [review]
Can you please resolve the conflicts :)

File: doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/11_Preparing_for_V11.md [hunk 2]
```diff
@@ -97,4 +97,6 @@
           target: 'symfony-config'
           options:
             directory: '/var/www/html/var/config/custom_reports'
-    ```
\ No newline at end of file
+    ```
+  
+- [Web2Print] Please keep in mind that deprecated `HeadlessChrome` will need to be replaced with the new processor `Chrome` when starting using 11.
```
Comment (mattamon) [inline doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/11_Preparing_for_V11.md]:
```suggestion
- [Web2Print] Please keep in mind that the deprecated processor `HeadlessChrome` needs to be replaced with the new processor `Chrome` in Pimcore 11.
```

**Review by mattamon — CHANGES_REQUESTED** [review]
LGTM, just changed the wording a bit to be more clear.

**Review by mattamon — APPROVED** [review]

**Pull request merged**